install.sh

```diff
@@ -405,6 +405,12 @@ nvm_do_install() {
     if nvm_profile_is_bash_or_zsh "${NVM_PROFILE-}"; then
       BASH_OR_ZSH=true
     fi
+    if ${BASH_OR_ZSH} && ! command grep -qc '$NVM_DIR/.nvm' "$NVM_PROFILE"; then
+      echo "=> Appending nvm path to $NVM_PROFILE"
+      command printf "${SOURCE_STR}" >> "$NVM_PROFILE"
+    else 
+      echo "=> nvm path already in ${NVM_PROFILE}"
+    fi
     if ! command grep -qc '/nvm.sh' "$NVM_PROFILE"; then
```

Comment:
can you elaborate a bit on this? it looks like this is basically just duplicating the block from lines 414-419.

Reply:
Oh yeah, I was thinking about the  path checking logic so just thought that we would check if the `NVM_DIR/.nvm` was in the profile for `BASH` and `ZSH`, we would avoid appending it else just append it. If i understood the issue correctly we would like to omit the addition of the path to the profile if it exists right ?

Reply:
Yes, but does line 414 not already do that?

Reply:
True how are we to then check the path ? I just thought that we would check it if its bash or zsh

Reply:
I guess I'm a bit confused. Why does it matter if it's bash or zsh? The original issue is implying that the current logic (which this PR does not change) is broken, and thus the "sourcing" lines are added more than once.

This implies that the existing check needs to be modified, rather than a new check added, for all shells.